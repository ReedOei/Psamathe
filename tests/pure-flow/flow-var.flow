;

10 --> var n : nat
4 --> var x : nat

// n |-> 10, x |-> 4

n[x] --> var m : nat
// n |-> 6, x |-> 4, m |-> 4
1 --> x
// n |-> 6, x |-> 5, m |-> 4
n[x] --> var k : nat
// n |-> 1, x |-> 5, m |-> 4, k |-> 5

[ any nat ; m, k ] --> var nums : list any nat
// n |-> 1, x |-> 5, m |-> 0, k |-> 0, nums |-> [ l1, l2 ], l1 |-> 4, l2 |-> 5

nums[4] --> var newNums : list any nat
// n |-> 1, x |-> 5, m |-> 0, k |-> 0, nums |-> [ l2 ], newNums |-> [l1], l1 |-> 4, l2 |-> 5

