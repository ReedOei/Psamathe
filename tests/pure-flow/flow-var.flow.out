<generatedTop>
  <common>
    <typeEnv>
      .Map
    </typeEnv>
    <types>
      .Map
    </types>
  </common>
  <k>
    .
  </k>
  <functions>
    .Map
  </functions>
  <lookup>
    k |-> selected ( loc ( 6 ) , loc ( 6 ) )
    m |-> selected ( loc ( 4 ) , loc ( 4 ) )
    n |-> selected ( loc ( 0 ) , loc ( 0 ) )
    newNums |-> selected ( loc ( 10 ) , loc ( 10 ) )
    nums |-> selected ( loc ( 7 ) , loc ( 7 ) )
    x |-> selected ( loc ( 2 ) , loc ( 2 ) )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( nat , ListItem ( 1 ) )
    loc ( 1 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 2 ) |-> resource ( nat , ListItem ( 5 ) )
    loc ( 3 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 4 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 5 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 6 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 7 ) |-> resource ( table ( .Vars ) any nat , ListItem ( loc ( 9 ) ) )
    loc ( 8 ) |-> resource ( nat , ListItem ( 4 ) )
    loc ( 9 ) |-> resource ( nat , ListItem ( 5 ) )
    loc ( 10 ) |-> resource ( table ( .Vars ) any nat , ListItem ( loc ( 12 ) ) )
    loc ( 11 ) |-> resource ( nat , ListItem ( 4 ) )
    loc ( 12 ) |-> resource ( nat , ListItem ( 4 ) )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
